src/social-network/shared.ts

```diff
@@ -49,11 +56,16 @@ export interface SocialNetworkWorkerAndUI {
      * if null, no public key detected.
      */
     publicKeyDecoder?: (text: string) => string | null
+    payloadEncoder?: (payload: string) => string
+    payloadDecoder?: (text: string) => string | null
```

Comment:
I think encode and decode should share the same type since they are inverse function of each other

Reply:
decode can be failed but encode can't...